### DISCUSSION TITLE
Generating Schema from all .graphqls files in resources directory

### DISCUSSION THREAD

**Discussion by boldalex** [graphql-java-kickstart/graphql-java-tools#733] [Q&A]
Hi guys,

Is it possible to use all .graphqls files from resources directory with SchemaParserBuilder. Something like this:
`SchemaParser.newParser().files("*.graphqls")...`

`SchemaParserBuilder.files()` method accepts array of files, yet the type of the parameter is String.
So when I try to include all the files as I mentioned above, I get this error:
`java.io.FileNotFoundException: classpath:*.graphqls`

Did anyone try to do this?
Thanks!

**Comment by zleonov** [comment]
Something like this will work: 
```
final Path resources = Paths.get(...);
final String[] filenames = Files.list(resources)
                                .map(Path::getFileName)
                                .map(Object::toString)
                                .filter(s -> s.endsWith(".graphqls")) // if there are other files in the directory
                                .toArray(String[]::new);
SchemaParser.newParser().files(filenames); // assuming they are on your classpath
```
**TL;DR,**
However, this is probably not what you want to do. The `SchemaParserBuilder.files()` method looks for the files on the classpath. Classpath resources are not necessarily synonymous with--and shouldn't be thought of as--files on the filesystem. They don't even have to be files, they could entries inside a JAR or on a remote server. Which is why the method arguments are `String`s and not `File`s or `Path`s.

As an example, if this is a webapp the above code would fail once you deploy it on an application server because the schema files would no longer be in that directory. Even if the directory path was relative, you wouldn't be able to reference it from inside the application server with jumping through hoops. It would depend on how your deployment artifacts are packaged.

If you really need this functionality because... I don't know... and you want your application to be portable, you would have to do something more interesting. For example, use Google Guava's `ClassPath` class to scan for all resources available from the `Thread.currentThread().getContextClassLoader()` and then filter them for GraphQL schema files.

Even doing that is some bad mojo. `ClassLoader`s are a never ending fun topic (sarcasm here) especially because application servers have to have their own custom `ClassLoader`s that often don't behave how you would expect from vanilla Java.

So in the end, just list the file names and make sure the files are on your classpath. How many schema files can there be?

:)
